module.nix

```diff
@@ -199,11 +224,10 @@ in
 
       services.nginx = mkIf cfg.nginx.enable {
         enable = true;
-        recommendedTlsSettings = true;
+        recommendedTlsSettings = lib.mkDefault (cfg.nginx.vhost.enableACME || (cfg.nginx.vhost.useACMEHost != null));
         recommendedProxySettings = true;
-        virtualHosts.${cfg.nginx.host} = {
-          inherit (cfg.nginx) enableACME;
-          forceSSL = cfg.nginx.enableACME;
+        virtualHosts.${cfg.nginx.host} = cfg.nginx.vhost // {
```

Comment:
This should really use `mkMerge`. Right now, definitions get silently discarded even when using `mkForce` (normally you'd get a "conflicting definitions"-error then). This goes against the semantics of the module system and would be extremely painful to debug when running into such an issue).